docs/DeveloperGuide.adoc

```diff
@@ -297,86 +299,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
-|=======================================================================
+|`* *` |user |do fuzzy search on problems |locate details of problems without having to go through the entire list
+
+|`* * *` |user |create custom tags |categorize problems via tags
+
+|`* * *` |user |add tags to problems |categorize problems via tags
+
+|`* *` |user |sort problems according to difficulty |locate problems easily
+
+|`* *` |user |add remarks to problems |have reference in the future
+
+|`* * *` |user |create plans containing problems |better prepare for interview
+
+|`* * *` |user |add problems to a plan |better prepare for interview
+
+|`* * *` |user |mark problems as complete/incomplete within plans |better prepare for interview
 
-_{More to be added}_
+|`* * *` |user |import database from <<json,JSON>> files |easily transfer data from one computer to another
+
+|`* * *` |user |export data into JSON format |easily transfer data from one computer to another
+
+|`* *` |advanced user |export data into CSV format |do some manipulation/processing on the data
+
+|=======================================================================
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `AlgoBase` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use Case 1: Add Problem
+
+*Guarantees*
+
+Problem will be added to the application if and only if the required fields are specified and the details entered are all valid.
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User launches AlgoBase.
+2.  User adds a new problem by entering the name of the problem, optionally specifying the description, author, weblink, source as well as any remarks or tags.
+3.  AlgoBase indicates successful addition of new problem.
+4.  AlgoBase displays details of problem added.
 +
 Use case ends.
 
 *Extensions*
+[none]
+* 2a. AlgoBase detects that an existing name already exists.
++
+[none]
+** 2a1. AlgoBase informs user that problem was not successfully added because the name already exists.
++
+Use case ends.
+
+* 2b. AlgoBase detects a missing name or invalid format for some field(s).
++
+[none]
+** 2b1. AlgoBase informs user that problem was not successfully added because the format is invalid.
++
+Use case ends.
+
+[discrete]
+=== Use Case 2: Edit Problem
+
+*Preconditions*
+
+1.  There are existing problems in the storage of AlgoBase.
+2.  The command is formatted correctly and the provided arguments meet the requirement for corresponding fields.
+
+*Guarantees*
+
+The matched problem in the application will be edited with the provided fields.
```

Comment:
I don't think this can be a guarantee, as your extension suggests that there may be failures.